### ISSUE TITLE
Port number 80 no more available (config. interface)

### ISSUE THREAD

**Issue by Wildernesss** [kzahel/web-server-chrome#202]
Hi,

In the last release of Chrome Web Server, it's no more possible to assign the port 80 to this web service (error message or constraint message : 'Enter a number between 1024 and 65535'). It's really embarrassing.

We use Chrome Web Server in a highschool in CS lessons into the web part and Chrome Web Server is really great for students to transform very quickly their computer into a small intranet web server. To avoid our students to type a too long url to test their web pages on this small real server context, we tell them to set port of Chrome Web Server to 80. In that way they can use regular url http://localhost and so on. It very natural, like in their real life and personal usage.

Asking them to put the port number at the end of each url is not very convenient and a little bit artificial. OK, ones could say it's the opportunity to learn what is the role of a port on a server. But even with port 80 we explain it.

In conclusion, it would be great if Chrome Web Server would still let user choose fully the number of the port and let him choose port 80.

Thanks a lot for your great project and reading

Regards

Laurent

**Comment by Wildernesss** [comment]
Just a complement: you can still set 80 ports but when you do this, the message 'Enter a number between 1024 and 65535' is then displayed in red just under the port number field. It's very confusing because the user can think that is new port number entry is rejected by Chrome Web Server that will not apply it (some students stops at this point and call teacher for help). But everything work fine if you don't take care of this message and use Chrome Web Server on port 80. Is it simply an error with this message that should simply not be displayed?

Laurent

**Comment by kzahel** [comment]
Oh. I didn't realize low numbered ports even worked.  I'll update to allow. Thank you for the detailed information on your use case

**Comment by kzahel** [comment]
I just tested on Windows and see that it DOES work on port 80 for windows. So I'm guessing your students are using windows?

**Comment by Wildernesss** [comment]
Yes, we work with Windows 10 environment. And yes it works with port 80 but the configuration panel of Chrome Web Server displays this red message warning when you enter 80 as port number: 'Enter a number between 1024 and 65535'

![image](https://user-images.githubusercontent.com/39602620/72878738-cc48c380-3cfb-11ea-8f79-0112471e2bfb.png)

**Comment by kzahel** [comment]
Sorry I dropped the ball on this one. I could change it to allow lower ports on windows, if that's still desirable.

**Comment by Wildernesss** [comment]
Hi Kyle. Yes, it still would be great. Chrome Web Server works well with 80 port but the error message is confusing for users that don't understand the message. Thanks again a lot for your great tool that we use every year with our students. Laurent

**Comment by manking109** [comment]
This does not work on port 80 at all in Chrome OS it will take 8080 but that will still show 8080 and not  a regular request from a brower that 80 is not needed.

**Comment by ethanaobrien** [comment]
@manking109 It does not work on port 80 on chrome os, but on linux and windows (idk about mac os) port 80 works
